users.js

```diff
@@ -640,6 +677,19 @@ class User {
 		}
 		return true;
 	}
+	lockName() {
+		let userid = this.userid;
+		for (let ip in this.ips) {
+			nameLockedIps[ip] = userid;
+		}
+		if (this.autoconfirmed) nameLockedUsers[this.autoconfirmed] = userid;
+		nameLockedUsers[this.userid] = userid;
+		this.namelocked = userid;
+		this.forceRename('Guest ' + this.guestNum, this.registered);
```

Comment:
`this.registered` should always be false. Usernames beginning with "Guest" can't be registered.